src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -559,21 +560,23 @@ trait Infer extends Checkable {
         val hasAny = pt :: restpe :: formals ::: argtpes ::: loBounds exists (_.dealiasWidenChain exists containsAny)
         !hasAny
       }
-      def argumentPosition(idx: Int): Position = context.tree match {
-        case x: ValOrDefDef => x.rhs match {
-          case Apply(fn, args) if idx < args.size => args(idx).pos
-          case _                                  => context.tree.pos
-        }
-        case _ => context.tree.pos
-      }
       if (settings.warnInferAny && context.reportErrors && canWarnAboutAny) {
-        foreachWithIndex(targs) ((targ, idx) =>
+        def argumentPosition(idx: Int): Position = context.tree match {
+          case x: ValOrDefDef => x.rhs match {
+            case Apply(fn, args) if idx < args.size => args(idx).pos
```

Comment:
I'm struggling to see how this is ever a useful position to report the warning (why is the n-th value parameter of an application related to the n-th type argument?), I'd suggest to drop this altogether.